spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+RSpec.describe "the ingredients index" do
+  before(:each) do
+    @recipe_1 = Recipe.create!(name: "Spaghetti", complexity: 3, genre: "Italian")
+    @recipe_2 = Recipe.create!(name: "Burger", complexity: 1, genre: "American")
+  end
+
+  it "shows the references recipe and all of its attributes" do
+    visit "/recipe/#{@recipe_1.id}"
+
+    expect(page).to have_content(@recipe_1.name)
+    expect(page).to have_content(@recipe_1.complexity)
+    expect(page).to have_content(@recipe_1.genre)
+  end
```

Comment:
This looks good. For second part of US 2:
```
As a visitor,
When I visit '/recipes/:id',
Then I see the recipe's name, complexity and genre,
and I see a list of the names of the ingredients for the recipe.
```
You will need to add `ingredients` and `RecipeIngredients` to your before each in order to be able to reach through `recipe` to get to its `ingredients`.